### DISCUSSION TITLE
Exposing a JSON field from SQLModel to Strawberry

### DISCUSSION THREAD

**Discussion by martarho** [strawberry-graphql/strawberry#1661] [Q&A] [answered]
Hi everyone,

I'm currently working with a postgresql database which contains a table with a JSONB field. I modelled it with SQLModel, and I'm trying to apply on top Strawberry to explose the data through GraphQL. 

The problem is dictionaries are not an accepted type on the graphql side. What's the recommended way one should deal with these cases? I'm currently learning both SQLModel and Strawberry, so if someone can show me an example it would be grand!

```
#
# Strawberry schema
#
class MyModel(SQLModel, table=True):
    id: int
    name: str
    jsonfield: dict = Field(sa_column=Column(JSON), default={"all": "true"})

@strawberry.experimental.pydantic.type(
    MyModel,
    fields=[
        "id",
       "name",
       "jsonfield"
    ],
)
class MyModelGQL:
    pass
```

**Comment by patrick91** [comment, marked as answer]
hey @martarho 😊 You'd need to create a custom scalar and then use that in your GraphQL type:

```python
JSON = scalar(
    NewType("JSON", object),  # mypy doesn't like `NewType("name", Any)`
    description="The `JSON` scalar type represents JSON values as specified by"
    " [ECMA-404](http://www.ecma-international.org"
    "/publications/files/ECMA-ST/ECMA-404.pdf).",
    specified_by_url="http://www.ecma-international.org"
    "/publications/files/ECMA-ST/ECMA-404.pdf",
    serialize=lambda v: v,
    parse_value=lambda v: v,
)

@strawberry.experimental.pydantic.type(
    MyModel
)
class MyModelGQL:
    id: auto
    name: auto
    jsonfield: JSON
```

note, we'll be adding the JSON scalar soon (hopefully later today) see: https://github.com/strawberry-graphql/strawberry/pull/1647/

**Reply by martarho** [reply]
Hi @patrick91 , thanks! <3 that makes sense! And thanks for the link, I'll watch that issue!
